### PR TITLE
Refactor code base

### PR DESCRIPTION
**Pull request by tortmayr**
[Refactor base package](https://github.com/eclipse-glsp/glsp-client/commit/753fdb86c741eacb31cbfb0ddd8d19e947aef859) 

- Move `FeedbackActionDispatcher` and related components into base package
- Refactor `FeedbackAwareModelCommand`. Remove `ModelRootChangeListener` API from this command and add it to the command stack instead. This ensures that the model listeners are notified after the update has been processed succeessfully. It also   makes it easier to rebind the `UpdateModelCommand` as you don't have to consider the listeners anymore.

[Update to lerna 7 and fix test cases](https://github.com/eclipse-glsp/glsp-client/commit/3ecb27506b2ae53f266d754b309a6113b1630e71)

[Refactor SelectionService](https://github.com/eclipse-glsp/glsp-client/commit/39ae849c29f14a44c6cdc521b25c0e4384b3e8ee) 

- Remove unnecessary Types.SelectionService binding and bind SelectionService directly instead
- Move `SelectionService` into base package. A lot of feature modules are dependent on this service => they would no longer work if the selectModule is not bound or customized

[Move container module out of base package](https://github.com/eclipse-glsp/glsp-client/commit/46ec6d8fd4b388aed489c7bb2b0000386732da99) 

The base package should have no direct dependencies to feature module to avoid circular dependency issues.

Fixes https://github.com/eclipse-glsp/glsp/issues/1049

### CODE REVIEW

**Review by martin-fleck-at — CHANGES_REQUESTED** [review]
I love a good refactoring and cleanup commit, so thank you very much Tobias! From my tests, everything seems to work well but I do have some questions and minor comments.

File: packages/client/src/base/di.config.ts [hunk 1]
```diff
@@ -59,9 +62,11 @@ const defaultGLSPModule = new ContainerModule((bind, _unbind, isBound, rebind) =
     bind(FocusTracker).toSelf().inSingletonScope();
     configureActionHandler(context, FocusStateChangedAction.KIND, FocusTracker);
 
-    // Model update initialization ------------------------------------
+    // Model update initialization ------------------------------------#
+    bind(TYPES.IFeedbackActionDispatcher).to(FeedbackActionDispatcher).inSingletonScope();
     configureCommand(context, FeedbackAwareUpdateModelCommand);
-    configureActionHandler(context, SetModelCommand.KIND, SetModelActionHandler);
+
+    rebind(MouseTool).to(RankingMouseTool).inSingletonScope();
```
[the remark this answers is not in the document]
Comment (tortmayr) [reply packages/client/src/base/di.config.ts:69]: I would rather make these methods protected. They were only public before due to the `IMouseTool` interface.

File: packages/client/src/base/view/mouse-tool.ts [hunk 2]
```diff
@@ -29,22 +24,22 @@ export interface IMouseTool {
 type MouseListenerMethods = keyof Omit<MouseListener, 'decorate'>;
 
 @injectable()
-export class RankingMouseTool extends MouseTool implements IMouseTool {
+export class RankingMouseTool extends MouseTool {
```
[the remark this answers is not in the document]
Comment (tortmayr) [reply packages/client/src/base/view/mouse-tool.ts:27]: I opt for changing the visibility of the public methods to private. As for the model parameter I think this was implemented this way to be flexible for future customization. While currently unused I think it's generally a good idea to forward all context arguments to the notify method. We can mark it as unused (`_model`) in our implementation. Potential adapters can than still access the model context if they need to.

File: packages/client/src/base/selection-service.ts [hunk 3]
```diff
@@ -240,3 +237,31 @@ export class SelectAllCommand extends Command {
         return context.root;
     }
 }
+
+export interface SelectFeedbackAction extends Omit<SelectAction, 'kind'>, Action {
```
[the remark this answers is not in the document]
Comment (tortmayr) [reply packages/client/src/base/selection-service.ts:241]: Right, if the selectModule is not bound nobody would handle the dispatched actions. Afaik sprotty will print a runtime warning for unhandled actions.
However, I think this is a fair trade off. If an adopter makes the conscious decision to remove the `selectModule` (and does not provide a custom replacement), they also have to take care of potentially dispatched select actions.
Either by 
- making sure that neither the client or server of their GLSP editor does send a SelectAction
- or binding fallback handlers for these actions

IMO this is still a way better compromise than what we had before were removing the `selectModule` would end up in corrupt inverify container and a not loading diagram,

**Comment by tortmayr** [comment]
Thanks for the fast review Martin. I have pushed an update to address the open issues.

File: packages/client/src/base/di.config.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (martin-fleck-at) [reply packages/client/src/base/di.config.ts:69]: Fine by me.

File: packages/client/src/base/view/mouse-tool.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (martin-fleck-at) [reply packages/client/src/base/view/mouse-tool.ts:27]: I see that you removed it, marking it as "unused" would have been fine by me as well.

File: packages/client/src/base/selection-service.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (martin-fleck-at) [reply packages/client/src/base/selection-service.ts:241]: Ok, that seems fair enough. I know that we have a similar message in the standalone case where we do not handle the dirty state. Thank you for the explanation!

**Review by martin-fleck-at — APPROVED** [review]
LGTM!

File: packages/client/src/base/view/mouse-tool.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (tortmayr) [reply packages/client/src/base/view/mouse-tool.ts:27]: After second consideration I saw that we were inconsistent anyways (e.g.the dispatchActions method did not have a reference to the model) so I opted for removing it

**Pull request merged**